ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/PeerTransactionTracker.java

```diff
@@ -69,6 +80,16 @@ public synchronized Set<Transaction> claimTransactionsToSendToPeer(final EthPeer
     }
   }
 
+  public synchronized Set<Transaction> claimTransactionHashesToSendToPeer(final EthPeer peer) {
+    final Set<Transaction> transactionHashesToSend = this.transactionHashesToSend.remove(peer);
+    if (transactionHashesToSend != null) {
+      markTransactionHashesAsSeen(peer, toHashList(transactionHashesToSend));
```

Comment:
should work, but i wonder if it would be more correct to mark them as seen after we know they've been sent to the peer. we could restore them to the cache in the exception handling of NewPooledTransactionHashesMessageSender:63

Reply:
good point, will review next